ci/taos/plugins-good/pr-prebuild-doxygen-tag.sh

```diff
@@ -151,15 +151,8 @@ function pr-prebuild-doxygen-tag(){
                             fi
 
                             # Check the doxygen tag written in upper case beacuase doxygen cannot use upper case tag such as '@TODO'.
-			    # This is a simple (incorrect) method to detect.
-			    # @todo it should be able to know if it is inside a comment or not!
-                            if [[ $line =~ [\ \t]*\*[\t \t]*"@"[A-Z] ]]; then
-                                echo "[ERROR] File name: $curr_file, $idx line, The doxygen tag sholud be written in lower case." >> $report_path
-                                check_result="failure"
-                                global_check_result="failure"
-                                latest_failed_file=$curr_file
-                            fi
-                            if [[ $line =~ //[\ \t]*"@"[A-Z] ]]; then
+                            # Let's check a comment statement that begins with '//','/*' or ' *'.
+                            if  [[ ($line =~ "@"[A-Z]) && ($line == "/*"* || $line == "//"* || $line == [[:blank:]]*"*"*)]]; then
```

Comment:
```==```? not ```=~```?

Reply:
I thought using `=~` can cause a problem when string has a value like this.
```c
str="...*...@DIM...";
```
So I tried to find a string which begins with `//,/*, *`

Reply:
The problem is that you cannot detect when they are indented:
```
$ line="    /* abc @D"
$ line2="/* abc @D"
$ if [[ $line == "/*"* ]]; then echo Y; else echo N; fi
N
$ if [[ $line2 == "/*"* ]]; then echo Y; else echo N; fi
Y
```

Reply:
Thank you for a detailed explanation.
I updated the format to detect indented comment!